classes/PodsI18n.php

```diff
@@ -0,0 +1,172 @@
+<?php 
+/**
+ * @package Pods
+ * @since 2.7
+ */
+class PodsI18n {
+
+	/**
+	 * @var PodsI18n
+	 */
+	static $instance = null;
+
+	/**
+	 * The strings
+	 */
+	public $strings = array();
+
+	/**
+	 * The localized strings
+	 */
+	public $localized = array();
+
+	/**
+	 * Singleton handling for a basic pods_i18n() request
+	 *
+	 * @return \PodsI18n
+	 *
+	 * @since 2.7
+	 */
+	public static function init() {
+
+		if ( ! is_object( self::$instance ) ) {
+			self::$instance = new PodsI18n();
+		}
+
+		return self::$instance;
+	}
+
+	/**
+	 * Constructur
+	 */
+	public function __construct() {
+		//self::init();
+		$this->localize_assets();
+	}
+
+	/**
+	 * Localize assets
+	 * 
+	 * @since 2.7
+	 */
+	public function localize_assets() {
+		// Create existing strings of this class
+		$this->strings = $this->create_strings();
+
+		/**
+		 * Add strings to the localization
+		 * Setting the key of your string to the original (non translated) value is mandatory
+		 * Note: Existing keys in this class will overwrite the ones of this filter!
+		 * 
+		 * @since 2.7
+		 * @see create_strings()
+		 * @param array
+		 * @return array format: 'Untranslated string' => 'Translated string with use of WP translate functions'
+		 */
+		$strings_extra = apply_filters( 'pods_localized_strings', array() );
+
+		$this->strings = array_merge( $strings_extra, $this->strings );
+
+		foreach ( $this->strings as $key => $str ) {
+			$this->register( $key, $str );
+		}
+
+		// Register our i18n script for JS
+		wp_register_script( 'pods-i18n', PODS_URL . 'ui/js/pods-i18n.js', array(), PODS_VERSION, true );
+
+		// Some other stuff we need to pass through
+		$localize = array(
+			'debug' => ( defined( 'SCRIPT_DEBUG' ) && SCRIPT_DEBUG == true ) ? true : false,
+		);
+		// Add localization to our i18n script
+		wp_localize_script( 'pods-i18n', 'pods_localized_strings', array_merge( $this->strings, $localize ) );
+	}
+
+	/**
+	 * Register function that creates the references and combines these with the translated strings
+	 * 
+	 * @since 2.7
+	 */
+	public function register( $string_key, $translation ) {
+		/**
+		 * Converts string into reference object variable
+		 * Uses the same logic as JS to create the same references
+		 */
+		$ref = '__' . $string_key;
+
+		// Add it to the strings localized
+		$this->strings[ $ref ] = $translation;
+		// Remove the old key
+		unset( $this->strings[ $string_key ] );
+	}
+
+	/**
+	 * Register our labels to use in JS
+	 * We need to register them as normal string to convert to JS references
+	 * And we need to register the translations to attach to these references, these may not be variables!
+	 * 
+	 * @since 2.7
+	 */
+	public function create_strings() {
+
+		return array(
+
+				'%s is required.' => 
```

Comment:
Oh I see now, it's used by the JS, this sounds better then in that case to provide JS a way to do __( .. ) type calls